lua/entities/gmod_tardis_interior/modules/sh_legacy.lua

```diff
@@ -0,0 +1,53 @@
+-- Legacy interior support
+if SERVER then
+	local function playerlookingat(self,ply,vec,fov,width)	
+		local disp = vec - self:WorldToLocal(ply:GetPos()+Vector(0,0,64))
+		local dist = disp:Length()
+		
+		local maxcos = math.abs( math.cos( math.acos( dist / math.sqrt( dist * dist + width * width ) ) + fov * ( math.pi / 180 ) ) )
+		disp:Normalize()
+		
+		if disp:Dot( ply:EyeAngles():Forward() ) > maxcos then
+			return true
+		end
+		
+		return false
+	end
+
+	ENT:AddHook("Use", "interior-legacy", function(self,a,c)
+		if self.metadata.ID=="legacy" and a:IsPlayer() and (not a:GetTardisData("outside")) and CurTime()>a:GetTardisData("outsidecool",0) then
+			local pos=Vector(0,0,0)
+			local pos2=self:WorldToLocal(a:GetPos())
+			local distance=pos:Distance(pos2)
+			if distance < 110 and playerlookingat(self,a,pos,10,10) then
+				self.exterior:PlayerThirdPerson(a,true)
+			end
+		end
+	end)
+end
+
+ENT:AddHook("Initialize", "interior-legacy", function(self)
+	if self.metadata.ID=="legacy" then
+		self.timerotor={}
+		self.timerotor.pos=0
+		self.timerotor.mode=1
+	end
+end)
+
+ENT:AddHook("Think", "interior-legacy", function(self)
+	if self.metadata.ID=="legacy" then
+		local moving = self.exterior:GetData("teleport",false)
+		local flightmode = self.exterior:GetData("flight",false)
+		if not CLIENT then return end
+		if (self.timerotor.pos>0 and not moving or flightmode) or (moving or flightmode) then
+		if self.timerotor.pos==1 then
+			self.timerotor.mode=0
+		elseif self.timerotor.pos==0 and (moving or flightmode) then
+			self.timerotor.mode=1
+		end
+		
+		self.timerotor.pos=math.Approach( self.timerotor.pos, self.timerotor.mode, FrameTime()*1.1 )
+		self:SetPoseParameter( "glass", self.timerotor.pos )
+		end
+	end
+end)
```

Comment:
Would be cool to have this a bit more officially defined as other interiors could well support this feature, and just have the legacy as one of them that uses this interior feature

Reply:
Other people use parts system for this so this is fine